tests/discounts/tests-discounts.php

```diff
@@ -393,6 +393,57 @@ public function test_discount_is_product_requirements_met() {
 		$this->assertTrue( self::$discount->is_product_requirements_met() );
 	}
 
+	/**
+	 * @covers edd_validate_discount
+	 */
+	public function test_edd_validate_discount_product_requirements_any() {
+		$products = array( self::$download->ID, 100 );
+		$args     = array(
+			'product_reqs'      => $products,
+			'product_condition' => 'any',
+			'max_uses'          => 10000,
+		);
+
+		edd_update_discount( self::$discount_id, $args );
+		$this->assertTrue( edd_validate_discount( self::$discount_id, $products ) );
+		$this->assertTrue( edd_validate_discount( self::$discount_id, array( self::$download->ID ) ) );
+		$this->assertTrue( edd_validate_discount( self::$discount_id, array( 100 ) ) );
```

Comment:
I'm feeling like the tests should also cover sending in a product which is _not_ part of the product requirements. But before I get too excited about adding in more assertions, I want to check with you, @cklosowski about it because on a recent PR you said we should limit the number of assertions per test. We do, however, have quite a few tests which have multiple (and even many) assertions, so my question is:
1. If the assertions cover an identical discount setup (like the discount should not change as in this method), is it okay to have multiple assertions?
2. I updated the discount object in this method and then again in both of the following methods, but I'm wondering if the modified discount is what's used in each. I wrote the update args assuming that was the case both ways.
3. It seems like the `test_discount_is_product_requirements_met` method should also test for an expected false return but it doesn't.

Anyway I would hypothetically like to add assertions for a hypothetical product which is not listed in the requirements, eg: 
```php
		$products[] = 3467;
		$this->assertFalse( edd_validate_discount( self::$discount_id, array( 3467 ) ) );
		$this->assertTrue( edd_validate_discount( self::$discount_id, $products ) );
```